### PR TITLE
tailscale-systray: init at 2022-10-19

### PR DESCRIPTION
**Pull request by qbit**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/misc/tailscale-systray/default.nix [hunk 1]
```diff
@@ -0,0 +1,28 @@
+{ pkgs, lib, buildGoModule, fetchFromGitHub, ... }:
```
Comment (linsui) [inline pkgs/applications/misc/tailscale-systray/default.nix:1]: Please add those packages as input directly.

Comment (qbit) [reply pkgs/applications/misc/tailscale-systray/default.nix]: done

**Comment by drupol** [comment]
URL: https://github.com/mattn/tailscale-systray

**Comment by drupol** [comment]
Why is this not building on `aarch64-darwin` and `x86_64-darwin` ?

**Comment by qbit** [comment]
My guess would be one of the deps doesn't build there.

**Comment by drupol** [comment]
```
Cannot nix-instantiate `tailscale-systray' because:
error: Package ‘libayatana-appindicator-0.5.92’ in /private/var/lib/ofborg/checkout/repo/38dca4e3aa6bca43ea96d2fcc04e8229/builder/macstadium-x86-44911362/pkgs/development/libraries/libayatana-appindicator/default.nix:33 is not supported on ‘x86_64-darwin’, refusing to evaluate.

```

**Comment by qbit** [comment]
Yep, confirmed - libayatana-appindicator doesn't build on darwin (`    platforms = platforms.linux;`).

I kinda have to wonder why one would want it on macos though, since the official client offers the same functionality.

**Review by drupol — APPROVED** [review]
LGTM!

**Comment by drupol** [comment]
How about adding:

```
  checkPhase = ''
    $GOPATH/bin/tailscale-systray
  '';
```?

**Comment by qbit** [comment]
The check would fail because it can't connect to the display.

**Comment by drupol** [comment]
Fair enough, LGTM anyway :)
So sad some options like `--accept-routes` are not supported...

**Review by 06kellyjac — CHANGES_REQUESTED** [review]

**Comment by drupol** [comment]
Just for the record, there's another systray utils for Tailscale, I'm using it since it has been merged in Nixpkgs, see it at: https://github.com/DeedleFake/trayscale

**Comment by 06kellyjac** [comment]
There's also a gnome extension for a talescale tray

**Comment by qbit** [comment]
Updated with suggestions, ty!

**Review by drupol — APPROVED** [review]

**Review by 06kellyjac — APPROVED** [review]
LGTM

If you need this in 23.05 LMK and we can put a label on this for backporting

**Pull request merged**